go-controller/pkg/ovn/pods.go

```diff
@@ -149,64 +183,99 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
+			podLogicalPorts[logicalSwitch] = fmt.Sprintf("%s_%s_%s", pod.Namespace, pod.Name, logicalSwitch)
+		}
 	}
+	podLogicalPorts[""] = fmt.Sprintf("%s_%s", pod.Namespace, pod.Name)
 
-	ipAddress := oc.getIPFromOvnAnnotation(pod.Annotations["ovn"])
+	for logicalSwitch, logicalPort := range podLogicalPorts {
+		out, stderr, err := util.RunOVNNbctlHA("--if-exists", "lsp-del",
+			logicalPort)
+		if err != nil {
+			logrus.Errorf("Error in deleting pod logical port "+
+				"stdout: %q, stderr: %q, (%v)",
+				out, stderr, err)
+		}
 
-	delete(oc.logicalPortCache, logicalPort)
+		var ipAddress string
+		if logicalSwitch == "" {
+			ipAddress = oc.getIPFromOvnAnnotation(pod.Annotations["ovn"])
+		} else {
+			networkAnntationMap := oc.getNetworkInfoFromOvnAnnotation(pod.Annotations["ovn_extra"], logicalSwitch)
+			ipAddress = oc.getIPFromOvnAnnotationMap(networkAnntationMap)
+		}
 
-	oc.lspMutex.Lock()
-	delete(oc.lspIngressDenyCache, logicalPort)
-	delete(oc.lspEgressDenyCache, logicalPort)
-	delete(oc.logicalPortUUIDCache, logicalPort)
-	oc.lspMutex.Unlock()
+		delete(oc.logicalPortCache, logicalPort)
 
-	if !oc.portGroupSupport {
-		oc.deleteACLDenyOld(pod.Namespace, pod.Spec.NodeName, logicalPort,
-			"Ingress")
-		oc.deleteACLDenyOld(pod.Namespace, pod.Spec.NodeName, logicalPort,
-			"Egress")
+		oc.lspMutex.Lock()
+		delete(oc.lspIngressDenyCache, logicalPort)
+		delete(oc.lspEgressDenyCache, logicalPort)
+		delete(oc.logicalPortUUIDCache, logicalPort)
+		oc.lspMutex.Unlock()
+
+		if !oc.portGroupSupport {
+			oc.deleteACLDenyOld(pod.Namespace, pod.Spec.NodeName, logicalPort,
+				"Ingress")
+			oc.deleteACLDenyOld(pod.Namespace, pod.Spec.NodeName, logicalPort,
+				"Egress")
+		}
+		oc.deletePodFromNamespaceAddressSet(pod.Namespace, ipAddress)
 	}
-	oc.deletePodFromNamespaceAddressSet(pod.Namespace, ipAddress)
+
 	return
 }
 
-func (oc *Controller) addLogicalPort(pod *kapi.Pod) {
+func (oc *Controller) addLogicalPort(pod *kapi.Pod, logicalSwitch string) string {
 	var out, stderr string
 	var err error
 	if pod.Spec.HostNetwork {
-		return
+		return ""
 	}
 
-	logicalSwitch := pod.Spec.NodeName
 	if logicalSwitch == "" {
 		logrus.Errorf("Failed to find the logical switch for pod %s/%s",
 			pod.Namespace, pod.Name)
-		return
+		return ""
 	}
 
 	if !oc.logicalSwitchCache[logicalSwitch] {
 		oc.logicalSwitchCache[logicalSwitch] = true
 		oc.addAllowACLFromNode(logicalSwitch)
 	}
 
-	portName := fmt.Sprintf("%s_%s", pod.Namespace, pod.Name)
+	var portName string
+	isDefaultPort := logicalSwitch == pod.Spec.NodeName
+
+	if isDefaultPort {
+		portName = fmt.Sprintf("%s_%s", pod.Namespace, pod.Name)
+	} else {
+		portName = fmt.Sprintf("%s_%s_%s", pod.Namespace, pod.Name, logicalSwitch)
+	}
+
 	logrus.Debugf("Creating logical port for %s on switch %s", portName, logicalSwitch)
 
-	annotation, isStaticIP := pod.Annotations["ovn"]
+	var ovnNetworkAnnotatedMap map[string]string
+	var isStaticIP bool
+	if isDefaultPort {
+		var annotation string
```

Comment:
no need to define the var here, just use := in the next line

Reply:
The next line is annotation, isStaticIP = pod.Annotations["ovn"]
and isStaticIP is needed outside of the 'if' scope.